lib/trie/trie.go

```diff
@@ -969,6 +969,9 @@ func (t *Trie) deleteBranch(branch *Node, key []byte) (
 	}
 
 	commonPrefixLength := lenCommonPrefix(branch.Key, key)
+	if (commonPrefixLength + 1) > len(key) {
+		return branch, false, 0
+	}
```

Comment:
`commonPrefixLength` is the common prefix length so it cannot be longer than the length of `key`. We can thus change the if condition to the simpler `commonPrefixLength == len(key)`.

I would suggest also assigning it to a `keyDoesNotExist` variable so that is servers as code documentation (or add a comment if you prefer), since it's not super obvious what that block does.

```suggestion
	keyDoesNotExist := commonPrefixLength == len(key)
	if keyDoesNotExist {
		return branch, false, 0
	}

```

Reply:
Yes, 2 == 2 makes more sense than (2 + 1) > 2.  Updated.